code/API_definitions/qos-booking.yaml

```diff
@@ -0,0 +1,1306 @@
+              schema:
+                $ref: "#/components/schemas/BookingInfo"
+              examples:
+                Booking_AVAILABLE:
+                  $ref: "#/components/examples/Booking_AVAILABLE"
+                Booking_AVAILABLE_WITHOUT_DEVICE:
+                  $ref: "#/components/examples/Booking_AVAILABLE_WITHOUT_DEVICE"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/GenericDevice404"
+        "422":
+          $ref: "#/components/responses/Generic422"
+        "429":
+          $ref: "#/components/responses/Generic429"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+      security:
+        - openId:
+            - "qos-booking:device-qos:read-by-device"
+
+components:
+  securitySchemes:
+    openId:
+      description: OpenID Connect authentication
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+    notificationsBearerAuth:
+      description: Bearer authentication for notifications
+      type: http
+      scheme: bearer
+      bearerFormat: "{$request.body#/sinkCredential.credentialType}"
+
+  parameters:
+    BookingId:
+      name: BookingId
+      in: path
+      description: Booking ID that was obtained from the createProvision operation
+      required: true
+      schema:
+        $ref: "#/components/schemas/BookingId"
+
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        type: string
+
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+
+  schemas:
+    BookingId:
+      description: Booking Identifier in UUID format
+      type: string
+      format: uuid
+
+    BaseBookingInfo:
+      description: Common attributes of a QoS Booking
+      type: object
+      properties:
+        device:
+          $ref: "#/components/schemas/Device"
+        qosProfile:
+          $ref: "#/components/schemas/QosProfileName"
+        applicationServer:
+          $ref: "#/components/schemas/ApplicationServer"
+        devicePorts:
+          description: The ports used locally by the device for flows to which the requested QoS profile should apply. If omitted, then the qosProfile will apply to all flows between the device and the specified application server address and ports
+          allOf:
+            - $ref: "#/components/schemas/PortsSpec"
+        applicationServerPorts:
+          description: A list of single ports or port ranges on the application server
+          allOf:
+            - $ref: "#/components/schemas/PortsSpec"
+        sink:
+          type: string
+          format: uri
```

Comment:
the format 'uri' allows for multiple uri schemes, such as 'http' (not secured HTTP) , 'mailto', etc. Might be good to restrict the uri-scheme to 'https'.